llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1543,6 +1543,21 @@ static void computeKnownBitsFromOperator(const Operator *I,
         }
         break;
       }
+
+      // Check for operations with the property that the magnitude of the
+      // result will never exceed that of the start value.
+      case Instruction::UDiv:
+      case Instruction::URem: {
```

Comment:
Isn't this missing a check for the operand?

(a udiv b) is <= a, but *not* <= b.

Reply:
Doesn't `matchSimpleRecurrence` set `R` to the start value? I've also checked for `RInst` to set the context correctly.